src/state/pools/index.ts

```diff
@@ -127,12 +140,17 @@ export const updateUserPendingReward = (sousId: string, account: string) => asyn
 
 export const fetchCakeVaultPublicData = () => async (dispatch) => {
   const publicVaultInfo = await fetchPublicVaultData()
-  dispatch(setCakeVaultPublicData(publicVaultInfo))
+  dispatch(setCakeVaultData(publicVaultInfo))
 }
 
 export const fetchCakeVaultFees = () => async (dispatch) => {
   const vaultFees = await fetchVaultFees()
   dispatch(setCakeVaultFees(vaultFees))
 }
 
+export const fetchCakeVaultUserData = (account: string) => async (dispatch) => {
+  const userData = await fetchVaultUser(account)
+  dispatch(setCakeVaultUser(userData))
+}
+
```

Comment:
These should be using `createAsyncThunk`. Check `state/prices/index.ts` for an example.

Reply:
Cheers man will dip back into these state tweaks tomorrow.

Reply:
Made that change bud if you could take a look? Also - why are asyncThunks preferred to just async functions that trigger a dispatch?